integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -207,3 +212,35 @@ async def resync_pipelines(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                         {**pipeline.asdict(), "__project": project.asdict()}
                         for pipeline in pipelines_batch
                     ]
+
+
+@ocean.on_resync(ObjectKind.MEMBER)
+async def resync_members(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    gitlab_resource_config: GitlabMembersResourceConfig = typing.cast(
+        GitlabMembersResourceConfig, event.resource_config
+    )
+    selector = gitlab_resource_config.selector
+
+    async def fetch_group_members(service, group):
+        if selector.enrich_with_public_email:
+            enriched_member_tasks = [
+                service.enrich_member_with_public_email(member)
+                async for members in service.get_all_group_members(group)
+                for member in members
+            ]
+            enriched_members = await asyncio.gather(*enriched_member_tasks)
+            return enriched_members
+        else:
+            member_dicts = [
+                member.asdict()
+                async for members in service.get_all_group_members(group)
+                for member in members
+            ]
+            return member_dicts
```

Comment:
why not using the `get_all_group_members`  once and then if `enrich_with_public_email` spawn requests for the members

Reply:
Great, the code will look like this then


`members = [
    member
    async for members_batch in service.get_all_group_members(group)
    for member in members_batch
]

if selector.enrich_with_public_email:
    enriched_member_tasks = [
        service.enrich_member_with_public_email(member)
        for member in members
    ]
    enriched_members = await asyncio.gather(*enriched_member_tasks)
    result = enriched_members
else:
    result = [member.asdict() for member in members]

return result`

Reply:
I'll refactor

Reply:
This is better for readability, thanks